src/Runner.Common/JobServerQueue.cs

```diff
@@ -44,6 +47,8 @@ public sealed class JobServerQueue : RunnerService, IJobServerQueue
         // queue for file upload (log file or attachment)
         private readonly ConcurrentQueue<UploadFileInfo> _fileUploadQueue = new();
 
+        private readonly ConcurrentQueue<ResultsUploadFileInfo> _resultsFileUploadQueue = new();
```

Comment:
I opted for its own queue as the results uploads will require quite a few more variables than the regular file/attachment upload (planid, jobid, stepid, etc)